log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileIO.java

```diff
@@ -12,39 +12,46 @@
 public class FileIO {
     private String file_name;
     private String directory;
-    private static final Charset charset = Charset.forName( "UTF-8" );
+    private static final Charset charset = Charset.forName("UTF-8");
 
     /**
      * Constructor
+     *
      * @param directory Directory relative to running instance
      * @param file_name File name
      */
-    public FileIO( String directory, String file_name ) {
+    public FileIO(String directory, String file_name) {
         this.file_name = file_name;
         this.directory = directory;
     }
 
     /**
      * Creates file if it doesn't exist
+     *
      * @return Creation success
      */
-    public boolean createFile() {
+    public synchronized boolean createFile() {
         try {
             if (!Files.exists(getFilePath())) {
-                Files.createDirectories( getDirectoryPath() );
+                Files.createDirectories(getDirectoryPath());
                 Files.createFile(getFilePath());
+                System.out.println( "Looks like file has been created.." );
+            }
+            if( !Files.exists(getFilePath()) ) {
+                MicroLogger.INSTANCE.log_Error( "[FileIO.createFile()] Failed to create a new file (", getFilePath(), ").");
```

Comment:
Shouldn't we instead append to the existing one? I.e. that's something I would expect from log - either create a file per session, or then append to a single file.